tests/functional/adapter/custom_schema_tests/test_custom_database.py

```diff
@@ -0,0 +1,112 @@
+import pytest
+import os
+from dbt.tests.util import (
+    check_relations_equal,
+    check_table_does_exist,
+    run_dbt
+)
+from tests.functional.adapter.custom_schema_tests.seeds import seed_queries
+
+
+_VIEW_1_SQL = """
+select * from {{ target.schema }}.seed
+""".lstrip()
+
+_VIEW_2_SQL = """
+{{ config(database='alt') }}
+select * from {{ ref('view_1') }}
+""".lstrip()
+
+_VIEW_3_SQL = """
+{{ config(database='alt', materialized='table') }}
+
+
+with v1 as (
+
+    select * from {{ ref('view_1') }}
+
+),
+
+v2 as (
+
+    select * from {{ ref('view_2') }}
+
+),
+
+combined as (
+
+    select last_name from v1
+    union all
+    select last_name from v2
+
+)
+
+select
+    last_name,
+    count(*) as count
+
+from combined
+group by 1
+""".lstrip()
+
+_CUSTOM_DB_SQL = """
+{% macro generate_database_name(database_name, node) %}
+    {% if database_name == 'alt' %}
+        {{ env_var('SNOWFLAKE_TEST_ALT_DATABASE') }}
+    {% elif database_name %}
+        {{ database_name }}
+    {% else %}
+        {{ target.database }}
+    {% endif %}
+{% endmacro %}
+""".lstrip()
+
+ALT_DATABASE = os.getenv("SNOWFLAKE_TEST_ALT_DATABASE")
+
+class TestOverrideDatabase:
+    @pytest.fixture(scope="class")
+    def schema(self):
+        return "custom_schema"
+
+    @pytest.fixture(scope="class")
+    def macros(self):
+        return {
+            "custom_db.sql": _CUSTOM_DB_SQL,
+        }
+
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "view_1.sql": _VIEW_1_SQL,
+            "view_2.sql": _VIEW_2_SQL,
+            "view_3.sql": _VIEW_3_SQL,
+        }
+
+    def test_snowflake_override_generate_db_name(self, project):
+        for query in seed_queries:
```

Comment:
nit: why not put this (and whatever else is preliminary) into a fixture called setUp?